precisely/iterable_matchers.py

```diff
@@ -139,4 +139,30 @@ def describe(self):
             ))
 
 
+def contains_only(matcher):
+    return IsSequenceContainsOnly(matcher)
+
+
+class IsSequenceContainsOnly(Matcher):
```

Comment:
I'd expect the name of the class to be `ContainsOnlyMatcher` (or whatever matches the name of the function).